core/src/main/java/org/infinispan/remoting/transport/jgroups/ClusterView.java

```diff
@@ -0,0 +1,61 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.infinispan.commons.util.Immutables;
+import org.infinispan.remoting.transport.Address;
+
+public class ClusterView {
+   static final int INITIAL_VIEW_ID = -1;
+   static final int FINAL_VIEW_ID = Integer.MAX_VALUE;
+
+   private final int viewId;
+   private final List<Address> members;
+   private final Set<Address> membersSet;
```

Comment:
the `List` can be removed in the future and replaced by a `Collection`. Most of our use case is to iterate or find live members. I only find a couple of uses cases where we fetch by index. I still think it is beneficial but I'm ok if you don't wanna do it now.